test/run-drun/throw-on-async-failure.mo

```diff
@@ -1,3 +1,4 @@
+//MOC-NO-FORCE-GC
```

Comment:
What is this new flag necessary?

Reply:
Several long-running tests failed with the incremental GC with sanity checks and/or `ic-ref-run`, with the typical error `Ingress message did not finish executing within 10000 batches, panicking`. 

The reason is that the `--force-gc` is always activated by `drun-wrapper` and the incremental GC has higher runtime complexity if always forced to run, especially in busy loops, e.g. in timer tests etc..

Reply:
**Update**: I adjusted the test run to use `--force-gc` explicitly. Hopefully, this simplifies it a little bit...